### PR TITLE
Fixes #8754: file ensure keys-> value cannot handle keys with special characters (like *)

### PR DESCRIPTION
**Pull request by ncharles**
https://www.rudder-project.org/redmine/issues/8754

### CODE REVIEW

File: tree/30_generic_methods/file_ensure_key_value.cf [hunk 1]
```diff
@@ -39,7 +39,7 @@ bundle agent file_ensure_key_value(file, key, value, separator)
   files:
       "${file}"
         create        => "true",
-        edit_line     => maintain_key_values("file_ensure_key_value.keys", "${separator}"),
+        edit_line     => ncf_maintain_keys_values("file_ensure_key_value.keys", "${separator}"),
```
Comment (peckpeck) [inline tree/30_generic_methods/file_ensure_key_value.cf]: Why do you replace key_values with keyS_values ?

Comment (ncharles) [inline tree/30_generic_methods/file_ensure_key_value.cf]: because it can set several keys

Comment (ncharles) [inline tree/30_generic_methods/file_ensure_key_value.cf]: see tree/30_generic_methods/file_ensure_keys_values.cf

**Comment by jooooooon** [comment]
Ping @peckpeck ? It would be good to get this in the release.

**Pull request merged**